test/extended/openstack/egressip.go

```diff
@@ -0,0 +1,475 @@
+
+	var networkClient *gophercloud.ServiceClient
+	var clientSet *kubernetes.Clientset
+	var err error
+	var infraID string
+	var workerNodeList *corev1.NodeList
+	var cloudNetworkClientset cloudnetwork.Interface
+	oc := exutil.NewCLI("openstack")
+
+	g.BeforeEach(func() {
+
+		g.By("Loading the kubernetes clientset")
+		clientSet, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		networkClient, err = client(serviceNetwork)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Error creating an openstack network client")
+
+		infrastructure, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		infraID = infrastructure.Status.InfrastructureName
+
+		g.By("Getting the worker node list")
+		workerNodeList, err = clientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "node-role.kubernetes.io/worker",
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.It("attached to a floating IP should be kept after EgressIP node failover with OVN-Kubernetes NetworkType", func() {
+
+		g.By("Getting the network type")
+		networkType, err := getNetworkType(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if networkType != NetworkTypeOVNKubernetes {
+			e2eskipper.Skipf("Test not applicable for '%s' NetworkType (only valid for '%s')", networkType, NetworkTypeOVNKubernetes)
+		}
+
+		// Skip based on number of required worker nodes
+		minWorkerNum := 2
+		g.By(fmt.Sprintf("Check the number of worker nodes (should be %d at least)", minWorkerNum))
+		e2e.Logf("Detected number of workers: '%d'", len(workerNodeList.Items))
+		if len(workerNodeList.Items) < minWorkerNum {
+			e2eskipper.Skipf("Skipping test as there are '%d' worker node(s) and minimum required is '%d'", len(workerNodeList.Items), minWorkerNum)
+		}
+
+		e2e.Logf("Getting 2 workers for the EgressIP node failover checks, aka primary and secondary workers")
+		primaryWorker := workerNodeList.Items[0]
+		secondaryWorker := workerNodeList.Items[1]
+		e2e.Logf("Selected primary EgressIP assignable worker: %s", primaryWorker.Name)
+		e2e.Logf("Selected secondary EgressIP assignable worker: %s", secondaryWorker.Name)
+
+		// Label the primary worker node as EgressIP assignable node
+		// "dummy" is added as value for the label key just because RemoveLabelOffNode is not
+		// able to remove the key if it doesn't have any value
+		g.By(fmt.Sprintf("Labeling the primary node '%s' with '%s'", primaryWorker.Name, egressAssignableLabelKey))
+		e2e.AddOrUpdateLabelOnNode(clientSet, primaryWorker.Name, egressAssignableLabelKey, "dummy")
+		defer e2e.RemoveLabelOffNode(clientSet, primaryWorker.Name, egressAssignableLabelKey)
+
+		g.By(fmt.Sprintf("Getting the EgressIP network from the '%s' annotation", egressIPConfigAnnotationKey))
+		egressIPNetCidrStr, err := getEgressIPNetwork(primaryWorker)
```

Comment:
This could be named getEgressIPAddress instead, but no strong opinion

Reply:
I think you're right, apparently this returns an address.

Reply:
This function is inspecting the node and getting the CIDR for the egress Network:
```
$ oc get node/ostest-h9vbm-worker-0-fkxdr -o json | jq .metadata.annotations | grep egress
  "cloud.network.openshift.io/egress-ipconfig": "[{\"interface\":\"e4340adc-fc03-4279-8fef-a5cfe4cbf5d0\",\"ifaddr\":{\"ipv4\":\"10.196.0.0/16\"},\"capacity\":{\"ip\":8}}]",
```

I think 'network' is more accurate than 'address'...